delphi-frontend/src/main/antlr3/au/com/integradev/delphi/antlr/Delphi.g

```diff
@@ -545,13 +546,17 @@ arrayIndices                 : lbrack (varType ','?)+ rbrack
 arrayElementType             : CONST<ConstArrayElementTypeNodeImpl>
                              | varType
                              ;
-setType                      : SET<SetTypeNodeImpl>^ OF varType
+fileType                     : FILE<FileTypeNodeImpl>^
                              ;
-fileType                     : FILE<FileTypeNodeImpl>^ (OF varType)?
+fullFileType                 : FILE<FileTypeNodeImpl>^ (OF varType)?
+                             ;
+setType                      : SET<SetTypeNodeImpl>^ OF varType
                              ;
 pointerType                  : '^'<PointerTypeNodeImpl>^ varType
                              ;
-stringType                   : STRING<StringTypeNodeImpl>^ (lbrack! expression rbrack!)?
+stringType                   : STRING<StringTypeNodeImpl>^
+                             ;
+fullStringType               : STRING<StringTypeNodeImpl>^ (lbrack! expression rbrack!)?
```

Comment:
Do you think it's worth removing the discard operators from `lbrack` and `rbrack` here, as part of the ongoing effort to remove these? Might make it tricky to do e.g. a quick fix that changes string types.

Does it also impact the range of the `StringTypeNode` (e.g. make it one character too short on the right, since the last token is removed)?

Reply:
This is a good call, I'll remove the discards.